esp-hal-common/src/spi.rs

```diff
@@ -200,41 +202,130 @@ where
 }
 
 #[cfg(feature = "eh1")]
-impl<T> embedded_hal_1::spi::ErrorType for Spi<T> {
-    type Error = Infallible;
-}
+pub use ehal1::*;
 
 #[cfg(feature = "eh1")]
-impl<T> embedded_hal_1::spi::nb::FullDuplex for Spi<T>
-where
-    T: Instance,
-{
-    fn read(&mut self) -> nb::Result<u8, Self::Error> {
-        self.spi.read_byte()
+mod ehal1 {
```

Comment:
Is this module necessary? I am not going to outright reject it, but I think this can all live with the rest of the code in this file.

I mainly ask as I intend to switch to these traits as the default as soon as a stable release is available and rely on `embedded-hal-compat` for supporting the older traits, so we would just end up pulling them out of this submodule anyways.

Reply:
It's not necessary at all, but I like to implement "feature gates" that way because IMO

1. It's easier to spot for developers which bits of the code belong to a feature
2. I don't have to spray single `#[cfg(feature = "eh1")]` all around the code

It's exactly equivalent to pulling the things out, though.

Reply:
Fair enough, just wanted to be sure of the motivation. We can just leave it as-is for now, thanks.